docs/core/classes.md

```diff
@@ -345,22 +308,20 @@ qx.Class.define("qx.test.Cat", {
   },
   members: {
     makeSound : function(x) {
-      this.constructor.someStaticMethod(x);
+      qx.test.Animal.someStaticMethod(x);
     }
   }
 });
 ```
 
 The syntax for accessing static variables simply is
-`this.constructor.someStaticVar`. Please note, for `this.constructor` to be
-available, the class must be a derived class of `qx.core.Object` , which is
-usually the case for regular, non-static classes.
-
-Instead of `this.constructor` you can also use the alternative syntax
-`this.self(arguments)`.
+`qx.test.Animal.someStaticVar`.  Please note that `this.constructor` is not
+recommended because it always refers to the subclass of the object, and therefore
+the implementation can change unexpectedly at run time.  Always specify class names
+explicitly in static methods.
```

Comment:
This is way too strong of a statement. `this.constructor` is, IMO, often the preferred form for one's own code because (a) it's far shorter, and (b) development is simpler, particularly when needing to change class names, because one needn't search/replace for all occurrences of the old class name. I would much prefer that we describe the problem (even leave the "not recommended" text if you want) as is done in the "Please note" sentence, but exclude the "Always" sentence.

Reply:
Do you have any suggestions?
@johnspackman : What do you think about?

Reply:
I'll amend it, but the reason why I put it in there is because there have been a couple of questions recently (in gitter and on SO) where users thought that `this.constructor` will return the class of the method in which it appears, and did not understand why it would not work when they extended or mixed in.